radix-engine/src/engine/process.rs

```diff
@@ -1239,31 +1261,37 @@ impl<'r, 'l, L: SubstateStore> Process<'r, 'l, L> {
         &mut self,
         resource_def_id: ResourceDefId,
         mint_params: MintParams,
-    ) -> Result<ResourceContainerState, RuntimeError> {
+    ) -> Result<ResourceContainer, RuntimeError> {
         let resource_def = self
             .track
             .get_resource_def_mut(&resource_def_id)
             .ok_or(RuntimeError::ResourceDefNotFound(resource_def_id))?;
         match mint_params {
             MintParams::Fungible { amount } => {
+                if !matches!(resource_def.resource_type(), ResourceType::Fungible { .. }) {
+                    return Err(RuntimeError::InvalidMintParams);
+                }
+
                 // Notify resource manager
-                resource_def
-                    .mint(&ResourceAmount::Fungible { amount })
-                    .map_err(RuntimeError::ResourceDefError)?;
+                resource_def.mint(amount);
 
                 // Allocate fungible
-                Ok(ResourceContainerState::fungible(
-                    amount,
+                Ok(ResourceContainer::new_fungible(
+                    resource_def_id,
                     resource_def.resource_type().divisibility(),
+                    amount,
                 ))
             }
             MintParams::NonFungible { entries } => {
+                if !matches!(
```

Comment:
Feels like this type checking should be done outside of `match mint_params` so that it's clear the check is made. Could we do this like:
```
impl MintParams {
    pub fn is_same_type(&self, resource_type: &ResourceType) -> bool {
        match self {
            Self::Fungible { .. } => matches!(resource_type, ResourceType::Fungible { .. }),
            Self::NonFungible { .. } => matches!(resource_type, ResourceType::NonFungible),
        }
    } 
 }
 ```

Reply:
Yes, will apply